R/pkg/R/SQLContext.R

```diff
@@ -41,10 +41,7 @@ infer_type <- function(x) {
   if (type == "map") {
     stopifnot(length(x) > 0)
     key <- ls(x)[[1]]
-    list(type = "map",
-         keyType = "string",
-         valueType = infer_type(get(key, x)),
-         valueContainsNull = TRUE)
+    paste0("map<string,", infer_type(get(key, x)), ">")
```

Comment:
One minor thing -- in the previous list we had an entry for `valueContainsNull` that we dont have any more. I can see that this was always `TRUE` so this probably doesn't affect functionality right now, but I am just wondering if we had it for some other purpose

cc @daveis